File: Data Structures & Algorithms/max-area-of-island/submission-0.go
func maxAreaOfIsland(grid [][]int) int {

	n := len(grid)
	if n == 0 {
		return 0
	}
	m := len(grid[0])
	maxArea := 0

	vis := make([][]bool, n)
	for i := range vis {
		vis[i] = make([]bool, m)
	}
    
	var dfs func(i, j int) int
    dfs = func(i, j int) int {

		 if i < 0 || i >= n || j < 0 || j >= m || 
           grid[i][j] == 0 || vis[i][j] {
            return 0
        }
        
        vis[i][j] = true
        area := 1            
        directions := [][2]int{{0, 1}, {0, -1}, {1, 0}, {-1, 0}}
        for _, dir := range directions {
            ni, nj := i + dir[0], j + dir[1]
            area += dfs(ni, nj)     
        }
        return area
    }

	for i := 0; i < n; i++{
		for j := 0; j < m; j++ {

			if !vis[i][j] && grid[i][j] == 1 {
				area := dfs(i, j)
				maxArea = max(area, maxArea)
			}
		}
	}
	return maxArea


}

File: Data Structures & Algorithms/longest-common-subsequence/submission-0.py
class Solution:
    def recurse(self, t1, t2, i, j):
        if i >= len(t1) or j >= len(t2):
            return 0

        if t1[i] == t2[j]:
            return 1 + self.recurse(t1, t2, i+1, j+1)
        else:
            return max(self.recurse(t1, t2, i, j+1), self.recurse(t1, t2, i+1, j))


    def longestCommonSubsequence(self, t1: str, t2: str) -> int:
        
        n, m = len(t1), len(t2)
        dp = [[0]*(m+1) for i in range(n+1)]

        for i in range(1, n+1):
            for j in range(1, m+1):
                # print(t1[i-1], t2[j-1], i, j)
                if t1[i-1] == t2[j-1]:
                    dp[i][j] = 1 + dp[i-1][j-1]
                else:
                    dp[i][j] = max(dp[i-1][j], dp[i][j-1])

        return dp[n][m]










